scipy/signal/_savitzky_golay.py

```diff
@@ -123,12 +123,7 @@ def savgol_coeffs(window_length, polyorder, deriv=0, delta=1.0, pos=None,
         x = x[::-1]
 
     order = np.arange(polyorder + 1).reshape(-1, 1)
-    if order.size == 1:
-        # Avoid spurious DeprecationWarning in numpy 1.8.0 for
-        # ``[1] ** [[2]]``, see numpy gh-4145.
-        A = np.atleast_2d(x ** order[0, 0])
-    else:
-        A = x ** order
+    A = x ** order
```

Comment:
Yep, you're right. Verified that in 1.8.2 this warning is gone. So change in this PR is okay.